coap-gateway/service/observation/deviceObserver.go

```diff
@@ -191,33 +207,36 @@ func NewDeviceObserver(ctx context.Context, deviceID string, coapConn ClientConn
 		notSyncedResources.Store(r.GetHref(), r)
 	}
 
-	if cfg.observationType == ObservationType_PerDevice {
-		resourcesObserver, err := createDiscoveryResourceObserver(ctx, deviceID, coapConn, callbacks, cfg.logger)
+	if cfg.ObservationType == ObservationType_PerDevice {
+		resourcesObserver, err := createDiscoveryResourceObserver(ctx, deviceID, coapConn, callbacks, cfg.Logger)
 		if err == nil {
 			return &DeviceObserver{
 				deviceID:          deviceID,
 				observationType:   ObservationType_PerDevice,
-				twinEnabled:       cfg.twinEnabled,
+				twinEnabled:       cfg.TwinEnabled,
 				rdClient:          rdClient,
 				resourcesObserver: resourcesObserver,
-				logger:            cfg.logger,
+				logger:            cfg.Logger,
 				raClient:          raClient,
 			}, nil
 		}
-		cfg.logger.Debugf("NewDeviceObserverWithResourceTwin: failed to create /oic/res observation for device(%v): %v", deviceID, err)
+		cfg.Logger.Debugf("NewDeviceObserver: failed to create /oic/res observation for device(%v): %v", deviceID, err)
+	}
+	if !cfg.ObservationPerResourceEnabled {
+		return nil, createError(fmt.Errorf("cannot create observation per resource for device(%v): disabled by configuration", deviceID))
```

Comment:
If developer googles for observation per resource, he won't find and understand what do we want from him.
Why not to name the property RequireBatchObserveEnabled? And inform user that `device(%v) doesn't  support batch observe, which is required by configuration`

Reply:
Is it observe or subscribe?

Reply:
added `requireBatchObserveEnabled`